File: Codechef/Existence/Existence.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main()
{
    int T;
    cin >> T;
    while (T--)
    {
        int X, Y, LHS, RHS;
        cin  >> X >> Y;
        LHS = pow(X, 4) + (4 * pow(Y, 2));
        RHS = 4 * (pow(X, 2) * Y);
        if (LHS == RHS)
        {
            cout << "YES" << endl;
        }
        else
        {
            cout << "NO" << endl;
        }
    }
    return 0;
}

File: Codechef/Existence/Existence.py
T = int(input())
while T:
    T -= 1
    X, Y = map(int, input().split())
    LHS = X**4 + (4 * Y**2)
    RHS = 4 * (X**2 * Y)
    if LHS == RHS:
        print("YES")
    else:
        print("NO")
